### PR TITLE
New resource `azurerm_ip_group_cidr`

### PR DESCRIPTION
**Pull request by tiwood**
This adds a new resource that allows us to manage the `CIDRs/IPs` in an existing `IP Group`

## Example Usage

```hcl
resource "azurerm_resource_group" "example" {
  name     = "test-rg"
  location = "West Europe"
}

resource "azurerm_ip_group" "example" {
  name                = "test-ipgroup"
  location            = azurerm_resource_group.test.location
  resource_group_name = azurerm_resource_group.test.name
}

resource "azurerm_ip_group_cidr" "example" {
  ip_group_id = azurerm_ip_group.example.id
  cidr        = "10.10.10.0/24"
}
```

CleanShot 2022-06-21 at 3 59 29

### CODE REVIEW

**Comment by tiwood** [comment]
@katbyte @tombuildsstuff
can some1 take a look? 

We need a way to add/remove from CIDR groups programmatically and might need to use a custom built provider if its unclear if this gets merged into the official provider.

File: internal/services/network/ip_group_resource.go [hunk 1]
```diff
@@ -47,8 +47,10 @@ func resourceIpGroup() *pluginsdk.Resource {
 			"resource_group_name": azure.SchemaResourceGroupName(),
 
 			"cidrs": {
-				Type:     pluginsdk.TypeSet,
-				Optional: true,
+				Type:       pluginsdk.TypeSet,
+				ConfigMode: pluginsdk.SchemaConfigModeAttr,
+				Optional:   true,
+				Computed:   true,
```
Comment (tombuildsstuff) [inline internal/services/network/ip_group_resource.go:53]: unfortunately this'd be a breaking configuration change, instead we'll need to revert this:

```suggestion
				Optional:   true,
```

and instead split the Create and Update methods here such that we retrieve the existing value for `cidrs` from the API and reuse this during the update, unless the `cidrs` block has changed, for example:

```
existing, err := client.Get(...)
if err != nil { .. }

update.Properties.Cidrs = existing.Properties.Cidrs
if d.HasChange("cidrs") {
    update.Properties.Cidrs = expandCidrs(d.Get("cidrs").([]interface{})
}
```

which'd allow users to make use of `ignore_changes` to ignore changes to the `cidrs` block on this resource, and use the separate resource instead - which'd need to be called out in the documentation for both of these resources

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @tiwood 

Thanks for this PR - apologies for the delayed review here.

I've taken a look through and whilst on the whole this is looking pretty good, we're going to need to fix the `azurerm_ip_group` resource such that we don't introduce a breaking change to users configurations here - I've left some notes inline about this.

However if we can fix this up then we should be able to take another look 👍 

Thanks!

**Review by katbyte — APPROVED** [review]
LGTM 🎫

**Comment by katbyte** [comment]
Superseded by https://github.com/hashicorp/terraform-provider-azurerm/pull/17321

**Pull request closed without merging**